# Import Libraries

In [1]:
import sqlite3
import pandas as pd

# Load Dataset

In [2]:
# Load the Excel dataset into a pandas DataFrame
df = pd.read_excel('Dataset for Data Analytics.xlsx')
df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


# Create Database

In [3]:
# Create SQLite database connection and load table
conn = sqlite3.connect('ecommerce.db')
df.to_sql('orders', conn, if_exists='replace', index=False)

1200

# SELECT Queries
This section displays records from the orders table using basic SELECT statements.

In [4]:
# This query displays all columns for the first 5 records
query1 = 'SELECT * FROM orders LIMIT 5;'
pd.read_sql_query(query1, conn)

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04 00:00:00,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23 00:00:00,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27 00:00:00,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15 00:00:00,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08 00:00:00,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [5]:
# This query selects specific columns (OrderID, Product, Quantity, TotalPrice) for 5 rows
query2 = 'SELECT OrderID, Product, Quantity, TotalPrice FROM orders LIMIT 5;'
pd.read_sql_query(query2, conn)

,OrderID,Product,Quantity,TotalPrice
0,ORD200000,Monitor,5,2853.10
1,ORD200001,Phone,2,302.70
2,ORD200002,Tablet,5,2753.40
3,ORD200003,Chair,1,273.19
4,ORD200004,Printer,4,2504.04


# WHERE Queries
This section filters table rows using the WHERE clause.

In [6]:
# This query filters rows to show orders with Delivered status
query3 = "SELECT OrderID, CustomerID, Product, TotalPrice, OrderStatus FROM orders WHERE OrderStatus = 'Delivered' LIMIT 5;"
pd.read_sql_query(query3, conn)

,OrderID,CustomerID,Product,TotalPrice,OrderStatus
0,ORD200004,C81840,Printer,2504.04,Delivered
1,ORD200015,C39416,Printer,473.96,Delivered
2,ORD200021,C99023,Monitor,1371.80,Delivered
3,ORD200024,C17470,Laptop,461.90,Delivered
4,ORD200025,C42577,Monitor,399.98,Delivered


In [7]:
# This query filters orders with a total price greater than 2500
query4 = 'SELECT OrderID, Product, Quantity, TotalPrice FROM orders WHERE TotalPrice > 2500 LIMIT 5;'
pd.read_sql_query(query4, conn)

,OrderID,Product,Quantity,TotalPrice
0,ORD200000,Monitor,5,2853.10
1,ORD200002,Tablet,5,2753.40
2,ORD200004,Printer,4,2504.04
3,ORD200010,Tablet,5,3129.85
4,ORD200032,Tablet,5,2683.60


# ORDER BY
This section sorts query output based on total price.

In [8]:
# This query sorts orders by TotalPrice in descending order to see highest sales
query5 = 'SELECT OrderID, Product, Quantity, TotalPrice FROM orders ORDER BY TotalPrice DESC LIMIT 5;'
pd.read_sql_query(query5, conn)

,OrderID,Product,Quantity,TotalPrice
0,ORD200789,Tablet,5,3456.40
1,ORD201122,Monitor,5,3390.95
2,ORD200632,Laptop,5,3390.80
3,ORD200469,Chair,5,3384.90
4,ORD200328,Tablet,5,3370.20


# GROUP BY
This section groups data by product category to count orders.

In [9]:
# This query counts total orders for each product category using COUNT()
query6 = 'SELECT Product, COUNT(*) AS TotalOrders FROM orders GROUP BY Product ORDER BY TotalOrders DESC;'
pd.read_sql_query(query6, conn)

,Product,TotalOrders
0,Printer,181
1,Tablet,179
2,Chair,178
3,Laptop,173
4,Desk,170
5,Monitor,163
6,Phone,156


# Aggregate Functions
This section uses aggregate functions SUM() and AVG() with GROUP BY.

In [10]:
# This query calculates total sales revenue for each payment method using SUM()
query7 = 'SELECT PaymentMethod, SUM(TotalPrice) AS TotalSales FROM orders GROUP BY PaymentMethod ORDER BY TotalSales DESC;'
pd.read_sql_query(query7, conn)

,PaymentMethod,TotalSales
0,Credit Card,263847.63
1,Online,262442.94
2,Cash,259786.29
3,Gift Card,246323.92
4,Debit Card,232361.18


In [11]:
# This query calculates average order total price for each product using AVG()
query8 = 'SELECT Product, AVG(TotalPrice) AS AveragePrice FROM orders GROUP BY Product ORDER BY AveragePrice DESC;'
pd.read_sql_query(query8, conn)

,Product,AveragePrice
0,Laptop,1110.558150
1,Chair,1098.989382
2,Printer,1080.732652
3,Monitor,1077.616012
4,Tablet,1042.284637
5,Desk,985.058412
6,Phone,972.579423


# Conclusion
In this project, I loaded an e-commerce dataset into SQLite using Python and Pandas. I wrote simple SQL queries using SELECT, WHERE, ORDER BY, GROUP BY, and aggregate functions (COUNT, SUM, AVG) to analyze sales data.

Key Observations:
- Printer had the highest order volume with 181 orders.
- Credit Card generated the highest total revenue ($263,847.63).
- Laptop had the highest average order price ($1110.56).